lib/cc/engine/analyzers/reporter.rb

```diff
@@ -0,0 +1,67 @@
+require 'cc/engine/analyzers/violation'
+
+module CC
+  module Engine
+    module Analyzers
+      class Reporter
+        def initialize(directory, language_strategy, io)
+          @directory = directory
+          @language_strategy = language_strategy
+          @io = io
+          @flay = Flay.new(flay_options)
```

Comment:
basically, it could be a `sexp_processor` that has a `process(sexp)` method on it, in case something later on takes the place of Flay